src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -72,16 +72,16 @@ internal class DurableClient : IDurableClient,
         string IDurableEntityClient.TaskHubName => this.TaskHubName;
 
         /// <inheritdoc />
-        HttpResponseMessage IDurableOrchestrationClient.CreateCheckStatusResponse(HttpRequestMessage request, string instanceId, bool returnInternalServerErrorOnFailure)
+        HttpResponseMessage IDurableOrchestrationClient.CreateCheckStatusResponse(HttpRequestMessage request, string instanceId, bool returnInternalServerErrorOnFailure, bool restartWithNewInstanceId)
```

Comment:
I don't know if adding a parameter to this API and the corresponding interfaces is worth it. We did this for `returnInternalServerErrorOnFailure` to ease the transition for people who needed the legacy behavior from Functions V1, but this is a new API, and adding parameters will break customer's unit tests.

I would say we just stick to a default behavior (new instance id?), and if customers want the other behavior they can tweak the URL returned.